### ISSUE TITLE
[bug] Package dbus-1 was not found in the pkg-config search path

### ISSUE THREAD

**Issue by jdu** [tauri-apps/tauri#4373]
### Describe the bug

Just saw the news about 1.0 and wanted to try out tauri after having not looked at it for a wile. Following along to the guide on this page https://tauri.app/v1/guides/getting-started/setup/html-css-js using `cargo tauri` and have hit a bit of a roadblock.

Running `cargo tauri dev` I get the following output:

<details>
<summary>Log Output</summary>

```
cargo tauri dev
   Compiling proc-macro-error v1.0.4
   Compiling lock_api v0.4.7
   Compiling libdbus-sys v0.2.2
   Compiling memoffset v0.6.5
   Compiling num-traits v0.2.15
   Compiling generic-array v0.14.5
   Compiling num-integer v0.1.45
   Compiling num-iter v0.1.43
error: failed to run custom build command for `libdbus-sys v0.2.2`

Caused by:
  process didn't exit successfully: `/home/jdu/Projects/pine/library/src-tauri/target/debug/build/libdbus-sys-6f56d2434582bb2f/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=DBUS_1_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=DBUS_1_STATIC
  cargo:rerun-if-env-changed=DBUS_1_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  pkg_config failed: `"pkg-config" "--libs" "--cflags" "dbus-1" "dbus-1 >= 1.6"` did not exit successfully: exit status: 1
  error: could not find system library 'dbus-1' required by the 'libdbus-sys' crate

  --- stderr
  Package dbus-1 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `dbus-1.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'dbus-1' found
  Package dbus-1 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `dbus-1.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'dbus-1' found

  One possible solution is to check whether packages
  'libdbus-1-dev' and 'pkg-config' are installed:
  On Ubuntu:
  sudo apt install libdbus-1-dev pkg-config
  On Fedora:
  sudo dnf install dbus-devel pkgconf-pkg-config

  thread 'main' panicked at 'explicit panic', /home/jdu/.cargo/registry/src/github.com-1ecc6299db9ec823/libdbus-sys-0.2.2/build.rs:16:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

</details>

I can confirm libdbus-1-dev and pkg-config are installed.

This is on Pop_Os which is essentially Ubuntu 22.04 under the hood. 

### Reproduction

1. On Pop_Os (Ubuntu 22.04)
2. Follow pre-requisites install (https://tauri.app/v1/guides/getting-started/prerequisites) for linux
3. `cargo install tauri-cli --version "^1.0.0"`
4. `cargo tauri init`
5. `cargo tauri dev` <- Error occurs here

### Expected behavior

I guess I was expecting a dev run of the application?

### Platform and versions

```shell
Environment
  › OS: Pop!_OS 22.04 X64
  › Node.js: 18.3.0
  › npm: 8.11.0
  › pnpm: Not installed!
  › yarn: Not installed!
  › rustup: 1.24.3
  › rustc: 1.61.0
  › cargo: 1.61.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
WARNING: no lock files found, defaulting to npm
  › @tauri-apps/cli [NPM]: 1.0.0
  › @tauri-apps/api [NPM]: Not installed!
  › tauri [RUST]: 1.0.0,
  › tauri-build [RUST]: 1.0.0,
  › tao [RUST]: 0.11.2,
  › wry [RUST]: 0.18.3,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../ui
  › devPath: ../ui
package.json not found

App directory structure
  ├─ ui
  └─ src-tauri

```
```


### Stack trace

```shell
cargo tauri dev
   Compiling proc-macro-error v1.0.4
   Compiling lock_api v0.4.7
   Compiling libdbus-sys v0.2.2
   Compiling memoffset v0.6.5
   Compiling num-traits v0.2.15
   Compiling generic-array v0.14.5
   Compiling num-integer v0.1.45
   Compiling num-iter v0.1.43
error: failed to run custom build command for `libdbus-sys v0.2.2`

Caused by:
  process didn't exit successfully: `/home/jdu/Projects/pine/library/src-tauri/target/debug/build/libdbus-sys-6f56d2434582bb2f/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=DBUS_1_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=DBUS_1_STATIC
  cargo:rerun-if-env-changed=DBUS_1_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  pkg_config failed: `"pkg-config" "--libs" "--cflags" "dbus-1" "dbus-1 >= 1.6"` did not exit successfully: exit status: 1
  error: could not find system library 'dbus-1' required by the 'libdbus-sys' crate

  --- stderr
  Package dbus-1 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `dbus-1.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'dbus-1' found
  Package dbus-1 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `dbus-1.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'dbus-1' found

  One possible solution is to check whether packages
  'libdbus-1-dev' and 'pkg-config' are installed:
  On Ubuntu:
  sudo apt install libdbus-1-dev pkg-config
  On Fedora:
  sudo dnf install dbus-devel pkgconf-pkg-config

  thread 'main' panicked at 'explicit panic', /home/jdu/.cargo/registry/src/github.com-1ecc6299db9ec823/libdbus-sys-0.2.2/build.rs:16:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

### Additional context

```
> sudo apt install libdbus-1-dev pkg-config
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libdbus-1-dev is already the newest version (1.12.20-2ubuntu4).
pkg-config is already the newest version (0.29.2-1ubuntu3).
0 upgraded, 0 newly installed, 0 to remove and 30 not upgraded.
```

**Comment by jdu** [comment]
```
❯ sudo apt install dbus-1
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package dbus-1
❯ sudo apt install libdbus-1
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libdbus-1
```

**Comment by jdu** [comment]
```
❯ sudo apt install libdbus-1-3
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libdbus-1-3 is already the newest version (1.12.20-2ubuntu4).
0 upgraded, 0 newly installed, 0 to remove and 30 not upgraded.
```

Seem to have the requirements for dbus-rs, not sure how to configure pkg-config to point to it.

**Comment by jdu** [comment]
Right, so on ubuntu 22.04 LTS dbus-1.pc gets installed to:

```
/usr/lib/x86_64-linux-gnu/pkgconfig
```

which I guess for some reason pkg-config just isn't set up to look in that folder, so configured `PKG_CONFIG_PATH` to this:

```
export PKG_CONFIG_PATH=/usr/lib/x86_64-linux-gnu/pkgconfig:$PKG_CONFIG_PATH
```

which has now reared a new problem:

<details>
<summary>Log Output</summary>

```
cargo tauri dev
   Compiling x11-dl v2.19.1
   Compiling libdbus-sys v0.2.2
   Compiling openssl-sys v0.9.74
   Compiling crossbeam-utils v0.8.9
   Compiling dirs-sys-next v0.1.2
   Compiling num_cpus v1.13.1
   Compiling filetime v0.2.16
   Compiling xattr v0.2.3
   Compiling shared_child v1.0.0
   Compiling os_pipe v1.0.1
   Compiling tempfile v3.3.0
   Compiling memoffset v0.6.5
   Compiling x11 v2.19.1
   Compiling bstr v0.2.17
   Compiling futures-channel v0.3.21
   Compiling futures-lite v1.12.0
   Compiling brotli v3.3.4
   Compiling generic-array v0.14.5
   Compiling png v0.17.5
   Compiling flate2 v1.0.24
   Compiling syn v1.0.96
   Compiling rand_core v0.5.1
   Compiling rand_core v0.6.3
   Compiling proc-macro-error-attr v1.0.4
   Compiling uuid v0.8.2
   Compiling uuid v1.1.2
   Compiling tar v0.4.38
   Compiling dirs-next v2.0.0
   Compiling tokio v1.19.2
   Compiling openssl v0.10.40
   Compiling native-tls v0.2.10
   Compiling parking_lot v0.12.1
   Compiling regex v1.5.6
   Compiling num-integer v0.1.45
   Compiling crossbeam-channel v0.5.5
   Compiling rustc_version v0.3.3
   Compiling rand_chacha v0.2.2
   Compiling rand_pcg v0.2.1
   Compiling rand_chacha v0.3.1
   Compiling block-buffer v0.10.2
   Compiling crypto-common v0.1.3
   Compiling dbus v0.9.5
   Compiling cfb v0.6.1
   Compiling num-iter v0.1.43
   Compiling globset v0.4.9
   Compiling field-offset v0.3.4
   Compiling rand v0.7.3
   Compiling rand v0.8.5
   Compiling digest v0.10.3
   Compiling proc-macro-error v1.0.4
   Compiling darling_core v0.13.4
   Compiling png v0.11.0
   Compiling infer v0.7.0
   Compiling ignore v0.4.18
   Compiling phf_generator v0.8.0
   Compiling notify-rust v4.5.8
   Compiling phf_generator v0.10.0
   Compiling sha2 v0.10.2
   Compiling ico v0.1.0
   Compiling phf_codegen v0.8.0
   Compiling string_cache_codegen v0.5.2
   Compiling selectors v0.22.0
   Compiling markup5ever v0.10.1
   Compiling serde_derive v1.0.137
   Compiling thiserror-impl v1.0.31
   Compiling futures-macro v0.3.21
   Compiling phf_macros v0.8.0
   Compiling cssparser v0.27.2
   Compiling html5ever v0.25.2
   Compiling cssparser-macros v0.6.0
   Compiling derive_more v0.99.17
   Compiling phf_macros v0.10.0
   Compiling ctor v0.1.22
   Compiling openssl-macros v0.1.0
   Compiling serialize-to-javascript-impl v0.1.1
   Compiling serde_repr v0.1.8
   Compiling darling_macro v0.13.4
   Compiling futures-util v0.3.21
   Compiling phf v0.8.0
   Compiling thiserror v1.0.31
   Compiling phf v0.10.1
   Compiling darling v0.13.4
   Compiling serde_with_macros v1.5.2
   Compiling serde v1.0.137
   Compiling futures-executor v0.3.21
   Compiling futures v0.3.21
   Compiling toml v0.5.9
   Compiling string_cache v0.8.4
   Compiling url v2.2.2
   Compiling serde_with v1.14.0
   Compiling serde_json v1.0.81
   Compiling semver v1.0.10
   Compiling serde_urlencoded v0.7.1
   Compiling os_info v3.4.0
   Compiling system-deps v6.0.2
   Compiling proc-macro-crate v1.1.3
   Compiling system-deps v5.0.0
   Compiling embed-resource v1.7.2
   Compiling cargo_toml v0.11.5
   Compiling treediff v3.0.2
   Compiling serialize-to-javascript v0.1.1
   Compiling attohttpc v0.19.1
   Compiling json-patch v0.2.6
   Compiling glib-macros v0.15.11
   Compiling soup2-sys v0.2.0
   Compiling javascriptcore-rs-sys v0.4.0
   Compiling gtk3-macros v0.15.4
   Compiling rfd v0.9.0
   Compiling glib-sys v0.15.10
   Compiling gobject-sys v0.15.10
   Compiling gdk-sys v0.15.1
   Compiling gio-sys v0.15.10
   Compiling atk-sys v0.15.1
   Compiling cairo-sys-rs v0.15.1
   Compiling gdk-pixbuf-sys v0.15.10
   Compiling pango-sys v0.15.10
   Compiling gtk-sys v0.15.3
   Compiling webkit2gtk-sys v0.18.0
   Compiling gdkx11-sys v0.15.1
   Compiling kuchiki v0.8.1
error: failed to run custom build command for `gdk-sys v0.15.1`

Caused by:
  process didn't exit successfully: `/home/jdu/Projects/pine/library/src-tauri/target/debug/build/gdk-sys-ae113870c36bd705/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=GDK_3.0_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  `"pkg-config" "--libs" "--cflags" "gdk-3.0" "gdk-3.0 >= 3.22"` did not exit successfully: exit status: 1
  error: could not find system library 'gdk-3.0' required by the 'gdk-sys' crate

  --- stderr
  Package xproto was not found in the pkg-config search path.
  Perhaps you should add the directory containing `xproto.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'xproto', required by 'xrender', not found

warning: build failed, waiting for other jobs to finish...
The following warnings were emitted during compilation:

warning: `"pkg-config" "--libs" "--cflags" "cairo" "cairo >= 1.14"` did not exit successfully: exit status: 1

error: failed to run custom build command for `cairo-sys-rs v0.15.1`

Caused by:
  process didn't exit successfully: `/home/jdu/Projects/pine/library/src-tauri/target/debug/build/cairo-sys-rs-a0441419533d730a/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=CAIRO_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=`"pkg-config" "--libs" "--cflags" "cairo" "cairo >= 1.14"` did not exit successfully: exit status: 1
  error: could not find system library 'cairo' required by the 'cairo-sys-rs' crate

  --- stderr
  Package xproto was not found in the pkg-config search path.
  Perhaps you should add the directory containing `xproto.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'xproto', required by 'xau', not found
```

</details>

So now onto seeing if I need some more deps and config...

**Comment by jdu** [comment]
Right, so I got there in the end...

This turned out to actually be because of `homebrew` which I had installed to be able to install a couple of tools which weren't available in the regular ubuntu apt repos. 

Apparently at some point it has installed it's own pkg-config and my PATH loaded that pkg-config before the system pkg-config which also meant for some reason my environment variable for `PKG_CONFIG_PATH` was empty. 

After unlinking, installing `libssl-dev` and restarting my shell, this seems to all be working now. 

Lesson learned - try not to mix package managers no matter what OS you're on.

Tauri is now building, i'll close this issue on this comment, but up to you lot to keep this issue in case someone else has the same issue somewhere down the line...
